api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/AvgPool3D.kt

```diff
@@ -0,0 +1,113 @@
+/*
+* Copyright 2020 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+* Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+*/
+
+package org.jetbrains.kotlinx.dl.api.core.layer.pooling
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.Layer
+import org.jetbrains.kotlinx.dl.api.core.layer.convolutional.ConvPadding
+import org.jetbrains.kotlinx.dl.api.core.shape.convOutputLength
+import org.jetbrains.kotlinx.dl.api.inference.keras.CHANNELS_FIRST
+import org.jetbrains.kotlinx.dl.api.inference.keras.CHANNELS_LAST
+import org.tensorflow.Operand
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+import org.tensorflow.op.nn.AvgPool3d
+
+/**
+ * Average pooling operation for 3D data (e.g. video, spatio-temporal).
+ *
+ * Downsamples the input by taking the average over a window of size [poolSize].
+ *
+ * @property [poolSize] Size of the pooling window.
+ * @property [strides] The amount of shift for pooling window in each pooling step. If
+ * `null`, it will default to [poolSize].
+ * @property [padding] Padding strategy; can be either of [ConvPadding.VALID] which means no
+ * padding, or [ConvPadding.SAME] which means padding the input equally such that the output
+ * has the same dimension as the input.
+ * @property [dataFormat] Data format of input; can be either of [CHANNELS_LAST] or [CHANNELS_FIRST].
+ */
+public class AvgPool3D(
+    public val poolSize: IntArray = intArrayOf(2, 2, 2),
+    public val strides: IntArray? = null,
```

Comment:
set the default value for the strides and convert both, strides and poolSize to the 5d array of longs

Reply:
Conversion was done, and I have discussed the benefit of using `null` as the default value for strides.